endpoints.go

```diff
@@ -33,3 +42,402 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
 	healthInfo := models.Health{Database: dbStatus, RabbitMQ: viper.GetString("dummy_health_rabbitmq")}
 	json.NewEncoder(w).Encode(healthInfo)
 }
+
+//Find survey by reference, short name, long name, or any combination of the three
+func getSurvey(w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    queryParams := r.URL.Query()
+    if len(queryParams) == 0 {
+        w.WriteHeader(http.StatusBadRequest)
+        errorString := models.Error{
+            Error: "No query parameters provided for search",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    paramSurveyRef := "survey_ref"
+    paramShortName := "short_name"
+    paramLongName := "long_name"
+
+    for params := range queryParams {
+        switch params {
+        case "surveyRef":
+            paramSurveyRef = queryParams.Get("surveyRef")
+        case "shortName":
+            paramShortName = queryParams.Get("shortName")
+        case "longName":
+            paramLongName = queryParams.Get("longName")
+        default:
+            w.WriteHeader(http.StatusBadRequest)
+            errorString := models.Error{
+                Error: "Invalid query parameter " + params,
+            }
+            json.NewEncoder(w).Encode(errorString)
+            return
+        }
+    }
+
+    queryString := "SELECT id, survey_ref, short_name, long_name, legal_basis, survey_mode FROM " + viper.GetString("db_schema") + ".survey WHERE survey_ref = $1 AND short_name = $2 AND long_name = $3"
```

Comment:
This means all of surveyRef, shortName and longName are required, which they shouldn't be. You'll need to build up the WHERE clause in the for/switch above to only include the parameters we're searching on